libs/shared/shared_common.c

```diff
@@ -252,7 +252,10 @@ _duplicate_program_descriptor(
     program_type_descriptor_copy = NULL;
 
 Exit:
-    ebpf_free(program_type_descriptor_copy);
+    if (program_type_descriptor_copy != NULL) {
+        ebpf_free((void*)program_type_descriptor_copy->name);
+        ebpf_free(program_type_descriptor_copy);
+    }
```

Comment:
Lines 320-324 also have multiple free lines.  To help avoid more such bugs in the future, I'd recommend creating an ebpf_program_type_descriptor_free() function that is called from here and there.